app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java

```diff
@@ -138,12 +138,19 @@ public static void onComponentCreated(Object conversionContext, AtomicReference<
             // Have to block the current thread until fetching is done
             // There's no known way to edit the text after creation yet
             RYDVoteData votingData;
+            long fetchStartTime = 0;
             try {
-                votingData = getVoteFetchFuture().get(MILLISECONDS_TO_BLOCK_UI_WHILE_WAITING_FOR_DISLIKE_FETCH_TO_COMPLETE, TimeUnit.MILLISECONDS);
+                Future<RYDVoteData> fetchFuture = getVoteFetchFuture();
+                if (SettingsEnum.DEBUG.getBoolean() && !fetchFuture.isDone()) {
+                    fetchStartTime = System.currentTimeMillis();
+                }
+                votingData = fetchFuture.get(MILLISECONDS_TO_BLOCK_UI_WHILE_WAITING_FOR_DISLIKE_FETCH_TO_COMPLETE, TimeUnit.MILLISECONDS);
             } catch (TimeoutException e) {
                 LogHelper.printDebug(() -> "UI timed out waiting for dislike fetch to complete");
+                recordTimeUISpentWaitingForNetworkCall(fetchStartTime);
                 return;
             }
+            recordTimeUISpentWaitingForNetworkCall(fetchStartTime);
```

Comment:
A try-catch-finally block can be used in order to call `recordTimeUISpentWaitingForNetworkCall` before return without using duplicate code.